src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1589,27 +1603,37 @@ void Partitioner::optimize(const std::string& func_name) {
             }
             ov::parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {
                 auto& funcall = func_group.refs[f_idx].get();
-                std::vector<ov::Tensor> to_concat;
+                std::vector<LazyTensor> to_concat;
+                // Fill tensor vector
                 for (auto&& cidx : to_concat_idx) {
-                    to_concat.push_back(funcall._closure[cidx]);
+                    to_concat.push_back(funcall._transformations[cidx]);
+                }
```

Comment:
This code was written with the idea that `ov::Tensor`s are implicitly shared (as internally they are PIMPLs). Does the same apply to `LazyTensor`s ?

Reply:
Done